From d698ccefa5c3a7d0d94b362f1ea366428463fd0e Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Tue, 1 Aug 2017 23:35:40 +0800
Subject: [PATCH 46/88] ARM: dts: sun8i: Add SY8106A regulator to Orange Pi 2

Add SY8106A regulator to r_i2c bus and enable the r_i2c bus on
Orange Pi 2, then set the power supply of the ARM cores to this
regulator, in order to enable DVFS.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 arch/arm/boot/dts/sun8i-h3-orangepi-2.dts | 26 ++++++++++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts b/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts
index ce4fae1bcba6..2996095cffc2 100644
--- a/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts
+++ b/arch/arm/boot/dts/sun8i-h3-orangepi-2.dts
@@ -114,6 +114,10 @@
 	status = "okay";
 };
 
+&cpu0 {
+	cpu-supply = <&reg_sy8106a>;
+};
+
 &de {
 	status = "okay";
 };
@@ -195,6 +199,28 @@
 	};
 };
 
+&r_i2c {
+	status = "okay";
+
+	reg_sy8106a: regulator@65 {
+		compatible = "silergy,sy8106a";
+		reg = <0x65>;
+		regulator-name = "vdd-cpux";
+		/*
+		 * The datasheet uses 1.1V as the minimum value of VDD-CPUX,
+		 * however the Armbian DVFS configuration to the BSP kernel
+		 * has some operating points under 1.1V and it's proven
+		 * to work well on all supported devices by Armbian.
+		 * As we adopted the Armbian DVFS OPP table, set the minimum
+		 * voltage to 1.0V here.
+		 */
+		regulator-min-microvolt = <1000000>;
+		regulator-max-microvolt = <1400000>;
+		regulator-boot-on;
+		regulator-always-on;
+	};
+};
+
 &reg_usb1_vbus {
 	gpio = <&pio 6 13 GPIO_ACTIVE_HIGH>;
 	status = "okay";
-- 
2.16.2

